components/more_channels/searchable_channel_list.jsx

```diff
@@ -174,6 +175,10 @@ export default class SearchableChannelList extends React.PureComponent {
                 id={`ChannelRow-${channel.name}`}
                 onClick={(e) => this.handleJoin(channel, e)}
                 tabIndex='0'
+                onMouseEnter={() => this.handleAtivateRowId(channel.id)}
+                onMouseLeave={this.handleDeactivateRowId}
+                onFocus={() => this.handleAtivateRowId(channel.id)}
+                onBlur={this.handleDeactivateRowId}
```

Comment:
It is generally not a good idea to change a components state based on the mouse events.
The way to go would be to leverage CSS pseudo classes to just show/hide elements.

example
```
.more-modal__row .more-modal__actions {
    display: none;
}
.more-modal__row:hover .more-modal__actions {
    display: block;
}
```

no state changes and re-rendering needed

Reply:
> the action area inside each element should not be rendered at all (display: none does not seem to suffice)

Thanks for the feedback but this was exactly the previous version. Based on your feedback above I changed the logic to use state to display/ hide action buttons. Please see my commit details https://github.com/mattermost/mattermost-webapp/pull/11262/commits/7d4ff7cb8a078db02e1773270c1346027fc2cf7f

Reply:
Yes ... I would actually prefer the CSS only approach ... lets re-do the CSS approach and I will have a look again if that was really the case here, ok?

Reply:
@michelengelen CSS approach is added back, please take a look 😄